test/recipes/80-test_cmp_http.t

```diff
@@ -275,18 +275,21 @@ sub start_mock_server {
     die "Invalid port: $server_port" unless $server_port =~ m/^\d+$/;
     my $pid = open($server_fh, "$cmd|") or die "Trying to $cmd";
     print "Pid is: $pid\n";
+    my $accept_msg = '';
     if ($server_port eq "0") {
         # Find out the actual server port
         while (<$server_fh>) {
             print;
             s/\R$//;                # Better chomp
-            next unless (/^ACCEPT/);
-            $server_port = $server_tls = $kur_port = $pbm_port = $1
-                if m/^ACCEPT\s.*?:(\d+)$/;
+            next unless (/^ACCEPT\s.*?:(.*)$/);
```

Comment:
This is no good, since there are situations where the server just prints "ACCEPT".
That's way I changed to a two-stage pattern check.
Yet even this is not robust enough.
To prevent waiting indefinitely in case the server does not print ACCEPT (e.g, on error),
we must not loop. Instead should read the first output line only, which should be the line wanted.

Reply:
>  To prevent waiting indefinitely in case the server does not print ACCEPT (e.g, on error),

But...  the server stops running in that case, no?  That should mean that $server_fh reaches eof and the loop stops.

In any case, I've changed it back to be that two-staged check

Reply:
Actually, the only problem I still see here is that in case the server does not print ACCEPT etc. the loop will the the tests stall.
So my main change request is not to loop or to exit the loop in any case after reading the first line.
The pattern itself is okay since we actually wanna match the `:` followed by a port number.
Even better would be `/^ACCEPT\s.*?:(\d+)` as it was before, then you can dispense with the later further matching.

Reply:
> Actually, the only problem I still see here is that in case the server does not print ACCEPT etc. the loop will the the tests stall.

Again, you said that this happens in case of error.  In that case, the sub-process should stop, no?  ...  and $server_fh should reach EOF, no?  I'm rather thinking that if `openssl cmp` in server mode fails to print ACCEPT and *still* continues to run, then there's a lack of error control *there*

Reply:
The reason that I do want to keep the loop is basically future proofing, in case we happen to print something else before the ACCEPT line.  We do exactly that for exactly that reason in TLSProxy.

Reply:
I understand that doing the loop gives more flexibility, but this bears the risk of waiting indefinitely.
Sure in case of a fatal server error it will terminate, so then this cannot happen.
Yet the server may simply misbehave,i.e., print nothing or print anything else and then wait for connections.
This is was recently happened again, for whatever exact reason(s), both with our CI runs and with NonStop,
and then the tests get stuck and it may be hard to find out why.

Reply:
I see no problem in simply requiring that the first thing that our HTTP test server emits is the ACCEPT.
This is exactly what I'm going to test independently before any other HTTP-based tests in #15592.
And if there is an error in the server and a respective output is produced before or instead of the ACCEPT,
this is still a good reason not to continue with the tests needing that server.

Reply:
> This is was recently happened again, for whatever exact reason(s), both with our CI runs and with NonStop, and then the tests get stuck and it may be hard to find out why.

It would actually not be very hard to see *if we were a bit more verbose!*
Then we could see what the started server actually outputs, and draw informed conclusions instead of guessing wildly.

Reply:
(yeah, I get quite exasperated when we're trying to deduce something without having the hard facts in front of us)

Reply:
I started deducing things simply because without using a NonStop system ourselves we cannot reproduce the issue ourselves, and it is more more less cumbersome and time-consuming to interact with those reporting those errors.
Adding more verbosity to the tests and its driver is certainly good.

Reply:
`make test TESTS=test_cmp_http V=1`

That's sufficient verbosity to get us the facts that we need.  I'm pretty sure that the NonStop folks can do this...  @rsbeckerca, can you confirm?

Reply:
While there is a risk when deducing things from fractional information and in fact part of my conclusions proved wrong,
I think most of them proved right. So I hope/believe this mode was better than saying nothing :wink:

Reply:
> `make test TESTS=test_cmp_http V=1`
> 
> That's sufficient verbosity to get us the facts that we need. I'm pretty sure that the NonStop folks can do this... @rsbeckerca, can you confirm?

Yes, upon my hints he did already use such commands.
What I suppose could be helpful beyond this if `80-test_cmp_http.t` echoed whatever it got from the server.

Reply:
Thank you.  Yes, there are indeed some details that needed fixing that you got my attention for,